js/whitebit.js

```diff
@@ -615,11 +707,440 @@ module.exports = class whitebit extends Exchange {
         return this.status;
     }
 
-    sign (path, api = 'publicV1', method = 'GET', params = {}, headers = undefined, body = undefined) {
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        let method = undefined;
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+            'side': side,
+            'amount': this.numberToString (this.amountToPrecision (symbol, amount)),
+        };
+        if (type === 'market') {
+            method = 'v4PrivatePostOrderMarket';
+            if (side === 'buy') {
+                let cost = this.safeNumber (params, 'cost');
+                const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
+                if (createMarketBuyOrderRequiresPrice) {
+                    if (price !== undefined) {
+                        if (cost === undefined) {
+                            cost = amount * price;
+                        }
+                    } else if (cost === undefined) {
+                        throw new InvalidOrder (this.id + " createOrder() requires the price argument for market buy orders to calculate total order cost (amount to spend), where cost = amount * price. Supply a price argument to createOrder() call if you want the cost to be calculated for you from price and amount, or, alternatively, add .options['createMarketBuyOrderRequiresPrice'] = false and supply the total cost value in the 'amount' argument or in the 'cost' extra parameter (the exchange-specific behaviour)");
+                    }
+                } else {
+                    cost = (cost === undefined) ? amount : cost;
+                }
+                const costToPrecision = this.costToPrecision (symbol, cost);
+                request['amount'] = this.numberToString (costToPrecision);
+            }
+        }
+        if (type === 'limit') {
+            method = 'v4PrivatePostOrderNew';
+            const convertedPrice = this.numberToString (this.priceToPrecision (symbol, price));
+            request['price'] = convertedPrice;
+        }
+        const customType = this.safeValue (params, 'type');
```

Comment:
Let's support `stockMarket`, `stopMarket`, `stopLimit` in the `type` argument without adding `customType`.

See how it's done here:

- https://github.com/ccxt/ccxt/blob/master/js/ftx.js#L1329
- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L2497
- https://github.com/ccxt/ccxt/blob/master/js/phemex.js#L1839

If the user has supplied `stopMarket` in the `type` argument

- if the user has supplied `stopPrice` or `activationPrice` in params – that takes precedence
- otherwise if the user supplied the `price` argument – that is used as the trigger activation price
- otherwise throw an exception to request a stop price from the user

Else if the `type` argument is `stopLimit`
    
- if the user has supplied `stopPrice` or `activationPrice` in params – that takes precedence
- otherwise throw an exception to request a stop price from the user
- use the `stopPrice` for the trigger activation price and the `price` for the limit price

Else ...